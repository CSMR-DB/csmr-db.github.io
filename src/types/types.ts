import React from 'react'

export type AnyExoticRefComponent<T> = React.ForwardRefExoticComponent<
  T & React.RefAttributes<any>
>

export type AnyExoticRefTargets =
  | ((instance: any) => void)
  | React.MutableRefObject<any>
  | null
