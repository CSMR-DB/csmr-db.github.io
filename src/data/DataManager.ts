import { ITool, toolsProvider } from './objects/toolsProvider'
import { themeProvider } from './objects/themeProvider'
import { DefaultTheme } from 'styled-components'
import { IRouteObject, routesProvider } from './objects/routesProvider'
import { ICourse, coursesProvider } from './objects/coursesProvider'
import { IContact, contactsProvider } from './objects/contactsProvider'
import {
  dynamicIconsProvider,
  IDynamicIcons,
} from './graphql/dynamicIconsProvider'
import {
  dynamicImagesProvider,
  IDynamicImages,
} from './graphql/dynamicImagesProvider'
import {
  experienceDataProvider,
  IExperienceData,
} from './graphql/experienceDataProvider'
import {
  gameWallpapersProvider,
  IGameWallpapers,
} from './graphql/gameWallpapersProvider'
import {
  ISeriesPosters,
  seriesPostersProvider,
} from './graphql/seriesPostersProvider'
import { ISiteMetadata } from '../types/interfaces'
import {
  ISkillsetData,
  skillsetDataProvider,
} from './graphql/skillsetDataProvider'
import { siteMetadataProvider } from './graphql/siteMetadataProvider'
import {
  IGraphicDesignData,
  graphicDesignDataProvider,
} from './graphql/graphicDesignDataProvider'
import {
  IPhotographySquares,
  photographySquaresProvider,
} from './graphql/photographySquaresProvider'
import {
  programmingDataProvider,
  IProgrammingData,
} from './graphql/programmingDataProvider'
import {
  photographyDataProvider,
  IPhotographyData,
} from './graphql/photographyDataProvider'

export abstract class AStaticDataManager {
  static contacts: IContact[]
  static courses: ICourse[]
  static routes: IRouteObject[]
  static theme: DefaultTheme
  static tools: ITool[]
  static siteMetadata: ISiteMetadata
  static skillsetData: ISkillsetData
  static experienceData: IExperienceData
  static programmingData: IProgrammingData
  static graphicDesignData: IGraphicDesignData
  static photographyData: IPhotographyData
  static gameWallpapers: IGameWallpapers
  static photographySquares: IPhotographySquares
  static seriesPosters: ISeriesPosters
  static dynamicImages: IDynamicImages
  static dynamicIcons: IDynamicIcons
}

/**
 * **Static Data Manager**
 *
 * Uses an abstract rather than an interface to implement, because TypeScript expects implemented interface to define the shape of an instance, not the class itself. Currently, using an abstract is the only way to get around that, as I can refer to it as
 * `typeof AStaticDataManager` anywhere I use it.
 *
 * Provides all 'static' data to Components through 1 single source of truth. No more duplicate-y GrahpQL queries with hidden differences between 2 files, but 1 predictable response.
 *
 * Everything else should remain consistent & predictable throughout the entire website to provide the best UX it can. I don't really need instances or a Singleton. The StaticDataManager shouldn't exist at runtime anyway.
 *
 * @Note:
 * StaticDataManager can only be used inside React Components, due to the use of the `useStaticQuery` hook in the Data properties.
 *
 * @FIXME:
 * I am not 100% satisfied with the current API "inconsistencies" when destructuring the data, but alas. That's a priority item for the next revision.
 */
export class StaticDataManager implements AStaticDataManager {
  static get routes(): IRouteObject[] {
    return routesProvider()
  }

  static get contacts(): IContact[] {
    return contactsProvider()
  }

  static get theme(): DefaultTheme {
    return themeProvider()
  }

  static get tools(): ITool[] {
    return toolsProvider()
  }

  static get courses(): ICourse[] {
    return coursesProvider()
  }

  static get siteMetadata(): ISiteMetadata {
    return siteMetadataProvider()
  }

  static get skillsetData(): ISkillsetData {
    return skillsetDataProvider()
  }

  static get experienceData(): IExperienceData {
    return experienceDataProvider()
  }

  static get programmingData(): IProgrammingData {
    return programmingDataProvider()
  }

  static get graphicDesignData(): IGraphicDesignData {
    return graphicDesignDataProvider()
  }

  static get photographyData(): IPhotographyData {
    return photographyDataProvider()
  }

  static get gameWallpapers(): IGameWallpapers {
    return gameWallpapersProvider()
  }

  static get photographySquares(): IPhotographySquares {
    return photographySquaresProvider()
  }

  static get seriesPosters(): ISeriesPosters {
    return seriesPostersProvider()
  }

  static get dynamicImages(): IDynamicImages {
    return dynamicImagesProvider()
  }

  static get dynamicIcons(): IDynamicIcons {
    return dynamicIconsProvider()
  }
}
