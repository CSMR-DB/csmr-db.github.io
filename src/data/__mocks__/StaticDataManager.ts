import { ExperienceData } from './../graphql/experienceDataProvider'
import { DynamicImages } from './../graphql/dynamicImagesProvider'
import { DynamicIcons } from './../graphql/dynamicIconsProvider'
import { ProjectsData } from '../graphql/projectsDataProvider'
import { SkillsetData } from '../graphql/skillsetDataProvider'
import { GameWallpapers } from '../graphql/gameWallpapersProvider'
import { SeriesPosters } from '../graphql/seriesPostersProvider'
import {
  SiteMetadata,
  SkillsetAllMarkdownRemark,
  ImageSharpAllFiles,
  ProjectsAllMarkdownRemark,
  ProjectMarkdownEdge,
  SkillsetMarkdownEdge,
  ExperienceAllMarkdownRemark,
} from '../../types/graphql.types'
import { PhotographySquares } from '../graphql/photographySquaresProvider'
import { PhotographyData } from '../graphql/photographyDataProvider'
import { Tool } from '../objects/toolsProvider'
import { AStaticDataManager } from '../StaticDataManager'
import { RouteObject } from '../objects/routesProvider'
import { Course } from '../objects/coursesProvider'
import { Contact } from '../objects/contactsProvider'
import { DefaultTheme } from 'styled-components'
import { mockSVGs } from './tools.mocks'

const mockImageCollection: ImageSharpAllFiles = {
  edges: [
    {
      node: {
        name: 'Photo',
        relativePath: '/photo.jpg',
        childImageSharp: {
          fluid: {
            aspectRatio: 1,
            sizes: '',
            src: '/photo.jpg',
            srcSet: '/photo.jpg',
          },
        },
      },
    },
  ],
}

const mockProgrammingSkillsetMdx: SkillsetAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'React',
          excerpt: 'I love JSX',
          favorite: true,
          icon: '',
          skillColor: 'red',
          time: 1337,
        },
        generatedRoute: '/skillset/react',
      },
    },
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'Jest',
          excerpt: 'It Jest needs to be done',
          favorite: true,
          icon: '',
          skillColor: 'blue',
          time: 1336,
        },
        generatedRoute: '/skillset/jest',
      },
    },
  ],
}

const mockProgrammingProjectsMdx: ProjectsAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'Portfolio',
          excerpt:
            'Gatsby, TypeScript, React, GraphQL, GSAP, Styled-Components, Jest, Enzyme',
          favorite: true,
          tags: [
            'Gatsby',
            'TypeScript',
            'React',
            'GraphQL',
            'GSAP',
            'Styled-Components',
            'Jest',
            'Enzyme',
          ],
        },
        generatedRoute: '/projects/portfolio',
      },
    },
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'Loopy',
          excerpt: 'I made a fantastic hula-loop once',
          favorite: true,
          tags: ['Hula', 'Loop', 'Perfection'],
        },
        generatedRoute: '/projects/portfolio',
      },
    },
  ],
}

const mockGraphicDesignProjectsMdx: ProjectsAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Graphic Design',
          date: '1994-01-08T00:00:00+0000',
          title: 'Circle',
          excerpt: 'I drew a fantastic circle once',
          favorite: true,
          tags: ['Pencil', 'Shapes', 'Drawing'],
        },
        generatedRoute: '/projects/circle',
      },
    },
  ],
}

const mockGraphicDesignSkillsetMdx: SkillsetAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Graphic Design',
          date: '1994-01-08T00:00:00+0000',
          title: 'Affinity Designer',
          excerpt: 'It has no proper geometry operations',
          favorite: false,
          icon: '',
          skillColor: 'green',
          time: 420,
        },
        generatedRoute: '/skillset/affinity_designer',
      },
    },
  ],
}

const mockExperienceMdx: ExperienceAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          title: 'React',
          backgroundColor: 'blue',
          type: 'Reactor',
          dateStart: '1994-01-08T00:00:00+0000',
          dateEnd: '1994-01-09T00:00:00+0000',
          excerpt: 'JSX is just fantastic',
        },
        generatedRoute: '',
      },
    },
    {
      node: {
        frontmatter: {
          title: 'Jest',
          backgroundColor: 'red',
          type: 'Jester',
          dateStart: '1994-01-08T00:00:00+0000',
          dateEnd: '1994-01-09T00:00:00+0000',
          excerpt: 'Jest a test',
        },
        generatedRoute: '',
      },
    },
  ],
}

export class StaticDataManager implements AStaticDataManager {
  static get theme(): DefaultTheme {
    return {
      palette: {
        dark: {
          normal: '#000',
          hover: '#222',
          active: '#444',
          lens: '#666',
        },
        light: {
          normal: '#fff',
          hover: '#ddd',
          active: '#bbb',
          lens: '#999',
        },
        first: { normal: '#f00', hover: '#d00', active: '#b00', lens: '#900' },
        second: {
          normal: '#0f0',
          hover: '#0d0',
          active: '#0b0',
          lens: '#090',
        },
        third: { normal: '#00f', hover: '#00d', active: '#00b', lens: '#009' },
        filter: {
          normal: 'saturation(4)',
          hover: 'saturation(4)',
          active: 'saturation(4)',
        },
      },
      breakpoints: {
        min: {
          smartphone: 'only screen and (min-width: 600px)',
          tablet: 'only screen and (min-width: 960px)',
          desktop: 'only screen and (min-width: 1200px)',
          ultrawide: 'only screen and (min-width: 1920px)',
        },
        max: {
          smartphone: 'only screen and (max-width: 600px)',
          tablet: 'only screen and (max-width: 960px)',
          desktop: 'only screen and (max-width: 1200px)',
          ultrawide: 'only screen and (max-width: 1920px)',
        },
      },
      fonts: {
        headings: 'Arial',
        copy: 'serif',
      },
    }
  }

  static get siteMetadata(): SiteMetadata {
    return {
      author: 'Jester',
      description: 'The TestRunner',
      image: '/',
      title: 'Jest',
      url: 'https://www.jest.testrunner',
    }
  }

  static get contacts(): Contact[] {
    return [
      {
        background: 'red',
        href: 'https://mock.test.nowhere',
        iconPath: '',
        text: 'Mock Option',
      },
    ]
  }

  static get courses(): Course[] {
    return [
      {
        category: 'Development',
        description: 'Experimenting with vanilla JS & jQuery',
        name: 'WebProgramming',
      },
    ]
  }

  static get routes(): RouteObject[] {
    return [
      {
        path: '/route',
        title: 'Route',
        boundKeys: ['R', 1],
      },
    ]
  }

  static get tools(): Tool[] {
    return [
      {
        name: 'React',
        svg: mockSVGs().RandomSampleSVG,
        color: 'red',
      },
    ]
  }

  static get photographyData(): PhotographyData {
    return {
      photographyData: mockImageCollection,
    }
  }

  static get photographySquares(): PhotographySquares {
    return {
      photographySquares: mockImageCollection,
    }
  }

  static get seriesPosters(): SeriesPosters {
    return {
      seriesPosters: mockImageCollection,
    }
  }

  static get gameWallpapers(): GameWallpapers {
    return {
      gameWallpapers: mockImageCollection,
    }
  }

  static get skillsetData(): SkillsetData {
    return {
      all: { ...mockProgrammingSkillsetMdx, ...mockGraphicDesignSkillsetMdx },
      top: {
        edges: [
          ...mockProgrammingSkillsetMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: SkillsetMarkdownEdge): boolean => favorite === true
          ),
          ...mockGraphicDesignSkillsetMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: SkillsetMarkdownEdge): boolean => favorite === true
          ),
        ],
      },
      allProgramming: mockProgrammingSkillsetMdx,
      allGraphicDesign: mockGraphicDesignSkillsetMdx,
    }
  }

  static get projectsData(): ProjectsData {
    return {
      all: { ...mockGraphicDesignProjectsMdx, ...mockProgrammingProjectsMdx },
      top: {
        edges: [
          ...mockGraphicDesignProjectsMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: ProjectMarkdownEdge): boolean => favorite === true
          ),
          ...mockProgrammingProjectsMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: ProjectMarkdownEdge): boolean => favorite === true
          ),
        ],
      },
      allProgramming: mockProgrammingProjectsMdx,
      allGraphicDesign: mockGraphicDesignProjectsMdx,
    }
  }

  static get experienceData(): ExperienceData {
    return {
      experienceData: mockExperienceMdx,
    }
  }

  static get dynamicIcons(): DynamicIcons {
    return {
      dynamicIcons: mockImageCollection,
    }
  }

  static get dynamicImages(): DynamicImages {
    return {
      dynamicImages: mockImageCollection,
    }
  }
}
