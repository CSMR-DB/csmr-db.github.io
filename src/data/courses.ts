import { theme } from './theme'

export interface ICourse {
  name: string
  description: string
  category:
    | 'Development'
    | 'Graphic Design'
    | 'Marketing'
    | 'Organizations'
    | 'Research'
    | 'Misc'
  color?: string
}

export const devCourses: ICourse[] = [
  {
    name: 'WebSkills',
    description: 'Just the absolute starter kit for working with HTML and CSS.',
    category: 'Development',
    color: theme.palette.dark.normal,
  },
  {
    name: 'Web Programming',
    description: `$('JavaScript'|'jQuery')`,
    category: 'Development',
    color: theme.palette.dark.normal,
  },
  {
    name: 'Realisation',
    description:
      'Using tools other than code to produce functional little apps. We worked with stuff like Appery.io and Layar.',
    color: theme.palette.dark.normal,
    category: 'Development',
  },
  {
    name: 'The Digital Workplace *',
    description:
      'A group project in about 8 groups of 4 to 5 students working together to create a prototype for a new Intranet for a client.',
    color: theme.palette.dark.normal,
    category: 'Development',
  },
  {
    name: 'Content Management Systems *',
    description: 'Building a website using Wordpress.',
    color: theme.palette.dark.normal,
    category: 'Development',
  },
]

export const marketingCourses: ICourse[] = [
  {
    name: 'Marketing & e-Business',
    description: 'An introduction course into digital marketing.',
    category: 'Marketing',
    // color: 'blue'
  },
  {
    name: 'Marketing & Communication',
    description: '',
    category: 'Marketing',
    // color: 'blue'
  },
  {
    name: 'Digital Methods',
    description:
      'Blogging about specific, current, topics regarding marketing.',
    category: 'Marketing',
    // color: 'blue'
  },
]

export const designCourses: ICourse[] = [
  {
    name: 'Workshop **',
    description:
      'We had 2 Workshops and these were not typical courses with an assessment, but instead were meant as an aid to other courses that relied on specific knowledge but were not designed to provide that in itself.',
    category: 'Graphic Design',
    // color: 'purple'
  },
  {
    name: 'Interaction Design',
    description:
      'Learning about core design principles like the 5-7 rule, visual aligment and more.',
    category: 'Graphic Design',
    // color: 'purple'
  },
  {
    name: 'Webdesign (with Axure)',
    description:
      'Apply what was taught at Interaction Design in a functional prototype built with Axure.',
    category: 'Graphic Design',
    // color: 'purple'
  },
  {
    name: 'Digital Storytelling',
    description:
      'Use your knowledge about Adobe Photoshop, Illustrator, After Effects and whatever else you want to produce an animated short.',
    category: 'Graphic Design',
    // color: 'purple'
  },
  {
    name: 'Film',
    description: 'Working with a camera, as opposed to just digital assets.',
    category: 'Graphic Design',
    // color: 'purple'
  },
]

export const organizationCourses: ICourse[] = [
  {
    name: 'Organisations & Communication',
    description: '',
    category: 'Organizations',
    // color: 'blue'
  },
  {
    name: 'Organisations & Change Management',
    description:
      'Say you need to implement a different workflow or new tools in an organization. How would you go about doing so?',
    category: 'Organizations',
    // color: 'blue'
  },
  {
    name: 'Processes & Information Supply',
    category: 'Organizations',
    description: '',
    // color: 'red'
  },
]

export const researchCourses: ICourse[] = [
  {
    name: 'Evaluation Research',
    description: 'Focus on qualitative research.',
    category: 'Research',
    // color: 'red'
  },
  {
    name: 'Testing & Measuring',
    description: 'Focus on quantitative research',
    category: 'Research',
    // color: 'red'
  },
]

export const miscCourses: ICourse[] = [
  {
    name: 'Doing New Things',
    description: '',
    category: 'Misc',
    // color: 'blue'
  },
  {
    name: 'Professional Skills',
    description: '',
    category: 'Misc',
    // color: 'blue'
  },
  {
    name: 'Legal Skills',
    description: '',
    category: 'Misc',
    // color: 'blue'
  },
  {
    name: 'Entrepreneurship',
    description: '',
    category: 'Misc',
    // color: 'blue'
  },
]

export const courses: ICourse[] = [
  ...devCourses,
  ...designCourses,
  ...marketingCourses,
  ...organizationCourses,
  ...researchCourses,
  ...miscCourses,
]
