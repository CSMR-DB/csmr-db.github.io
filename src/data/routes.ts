export interface IRouteObject {
  path: string
  title: string
}

export const routes: IRouteObject[] = [
  {
    path: '/about',
    title: 'About',
  },
  {
    path: '/skillset',
    title: 'Skillset',
  },
  {
    path: '/projects',
    title: 'Projects',
  },
  {
    path: '/contact',
    title: 'Contact',
  },
]
