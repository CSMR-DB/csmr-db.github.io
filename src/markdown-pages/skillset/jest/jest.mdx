---
category: 'Programming'
level: 2
skillColor: '#C63D14'
time: 400
title: 'Jest'
excerpt: "I mean, it's one of many testing frameworks. I mainly just like the logo."
icon: <JestSVG/>
---
