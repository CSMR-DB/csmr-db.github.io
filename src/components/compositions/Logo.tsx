import React from 'react'
import styled, {
  StyledComponent,
  css,
  FlattenSimpleInterpolation,
} from 'styled-components'
import {
  BackgroundColorProperty,
  BorderProperty,
  StrokeProperty,
  WidthProperty,
} from 'csstype'

import { theme } from '../../data/theme'

interface ILogoProps {
  size: WidthProperty<1>
  strokeColor?: string
  strokeSize?: number
  fill: string
  background?: BackgroundColorProperty
  bordered?: boolean
}

const StyledLogoBase: StyledComponent<
  'div',
  any,
  Partial<ILogoProps>,
  never
> = styled.div`
  float: left;
  border-radius: 999rem;
  display: block;
  z-index: 999;

  border: ${(props: Partial<ILogoProps>): BorderProperty<1> =>
    props.bordered ? `1px solid ${theme.primary}` : 'none'};

  & svg {
    float: left;
  }
`

interface ILogoPathProps {
  strokeWidth: number
  stroke: StrokeProperty
}

const StyledPath: StyledComponent<
  'path',
  any,
  ILogoPathProps,
  never
> = styled.path`
  ${({
    stroke = 'white',
    strokeWidth = 0,
  }: ILogoPathProps): FlattenSimpleInterpolation => css`
    stroke: ${stroke};
    stroke-width: ${strokeWidth};
  `}
`

export const Logo: (props: ILogoProps) => JSX.Element = ({
  size,
  strokeColor = 'transparent',
  strokeSize = 0,
  fill,
  background,
}: ILogoProps): JSX.Element => (
  <StyledLogoBase background={background}>
    <svg
      width={size}
      height={size}
      viewBox="0 0 270.93333 270.93333"
      version="1.1"
      id="svg3719"
    >
      <g id="layer1" transform="translate(0,-26.06667)">
        <g
          transform="matrix(0.35277777,0,0,-0.35277777,11.994447,373.2)"
          id="g3729"
        >
          <StyledPath
            stroke={strokeColor}
            strokeWidth={strokeSize}
            fill={fill}
            d="M 500,850.00049 C 444.771,850.00049 399.99951,805.22901 399.99951,750 h 0.001 V 600.10694 c -3e-5,-0.0358 -0.001,-0.0712 -0.001,-0.10694 V 536.61914 C 385.2906,545.12797 368.21496,550.00048 350,550.00048 294.77099,550.00048 249.99951,505.229 249.99951,450 c 0,-55.22901 44.77148,-100.00049 100.00049,-100.00049 55.229,0 100.00049,44.77148 100.00049,100.00049 v 63.38086 C 464.7091,504.87148 481.78571,499.99951 500,499.99951 c 55.22901,0 99.99903,44.77149 99.99903,100.00049 0,29.87 -13.0988,56.676 -33.8628,75 20.763,18.324 33.86426,45.13 33.86426,75 0,55.22901 -44.77148,100.00049 -100.00049,100.00049 z m 0,-50.00097 c 27.57,0 49.99952,-22.42951 49.99952,-49.99952 0,-27.57 -22.42952,-49.99951 -49.99952,-49.99951 v -50.00097 c 27.571,0 49.99952,-22.42952 49.99952,-49.99952 0,-27.57 -22.42852,-49.99952 -49.99952,-49.99952 -27.55084,0 -49.96837,22.3987 -49.99951,49.94239 v 0.0571 63.41309 23.17383 63.41308 c 0,27.57001 22.42951,49.99952 49.99951,49.99952 z M 199.99999,550.00048 C 144.77099,550.00048 99.999509,505.229 99.999509,450 c 0,-55.22901 44.771481,-100.00049 100.000481,-100.00049 22.666,0 43.56639,7.54486 60.33838,20.25586 -11.28,12.673 -19.91564,27.75058 -25.03564,44.37158 -9.052,-9.033 -21.53574,-14.62647 -35.30274,-14.62647 -27.57,0 -49.99951,22.42952 -49.99951,49.99952 0,27.57 22.42951,49.99951 49.99951,49.99951 13.767,0 26.25074,-5.59347 35.30274,-14.62647 5.12,16.62101 13.75564,31.69859 25.03564,44.37159 -16.77199,12.711 -37.67238,20.25585 -60.33838,20.25585 z M 350,499.99951 c 27.57,0 49.99951,-22.42951 49.99951,-49.99951 0,-27.57 -22.42951,-49.99952 -49.99951,-49.99952 -27.57,0 -49.99952,22.42952 -49.99952,49.99952 0,27.57 22.42952,49.99951 49.99952,49.99951 z"
            id="path3777"
          />
        </g>
      </g>
    </svg>
  </StyledLogoBase>
)
