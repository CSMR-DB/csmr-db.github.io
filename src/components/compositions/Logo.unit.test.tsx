import React, { Component } from 'react'
import { ThemeProvider } from 'styled-components'
import { create, ReactTestRenderer } from 'react-test-renderer'
import { mount, ReactWrapper } from 'enzyme'

import { StaticDataManager } from '../../data/StaticDataManager'

import { Logo } from './Logo'
import { RTRTree } from '../../types/jest.types'

describe('Logo', (): void => {
  it('should render correctly', (): void => {
    const comp: ReactTestRenderer = create(
      <ThemeProvider theme={StaticDataManager.theme}>
        <Logo />
      </ThemeProvider>
    )

    expect(comp).toMatchSnapshot()
  })

  it('should have the following static style rules', (): void => {
    const comp: ReactTestRenderer = create(
      <ThemeProvider theme={StaticDataManager.theme}>
        <Logo />
      </ThemeProvider>
    )
    const json: RTRTree = comp.toJSON()

    expect(comp).toMatchSnapshot()

    expect(json).toHaveStyleRule('float', 'left')
    expect(json).toHaveStyleRule('border-radius', '999rem')
    expect(json).toHaveStyleRule('display', 'block')
    expect(json).toHaveStyleRule('z-index', '999')
  })

  it('should have the following configurable style rules when configured', (): void => {
    const comp: ReactTestRenderer = create(
      <ThemeProvider theme={StaticDataManager.theme}>
        <Logo
          $size={'64rem'}
          $strokeColor={StaticDataManager.theme.palette.second.normal}
          $strokeWidth={8}
        />
      </ThemeProvider>
    )
    const json: RTRTree = comp.toJSON()

    expect(comp).toMatchSnapshot()

    // expect(json).toHaveStyleRule('border', `1px solid ${StaticDataManager.theme.palette.first.normal}`)
    expect(json).toHaveStyleRule('width', '64rem', {
      modifier: 'svg',
    })
    expect(json).toHaveStyleRule('height', '64rem', {
      modifier: 'svg',
    })
    expect(json).toHaveStyleRule('stroke-width', '8', {
      modifier: 'svg > g > path',
    })
    expect(json).toHaveStyleRule(
      'stroke',
      StaticDataManager.theme.palette.second.normal,
      {
        modifier: 'svg > g > path',
      }
    )
  })

  it('should render internally', (): void => {
    const comp: ReactWrapper<any, Readonly<{}>, Component<{}, {}, any>> = mount(
      <ThemeProvider theme={StaticDataManager.theme}>
        <Logo />
      </ThemeProvider>
    )

    expect(comp.find('path').length).toEqual(2)
  })
})
