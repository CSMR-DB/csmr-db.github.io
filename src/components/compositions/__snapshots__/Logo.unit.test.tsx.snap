// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should have the following configurable style rules when configured 1`] = `
.c0 {
  float: left;
  border-radius: 999rem;
  display: block;
  z-index: 999;
  background: transparent;
}

.c0 svg {
  float: left;
}

.c0 svg {
  width: 64rem;
  height: 64rem;
}

.c0 svg > g > path {
  stroke: #0f0;
  stroke-width: 8;
}

<div
  className="c0"
>
  <svg
    viewBox="0 0 256 256"
  >
    <g
      transform="matrix(1,0,0,1,0,-320)"
    >
      <path
        d="M99.143,406.249C104.26,395.454 115.266,388 128,388C145.661,388 160,402.339 160,420C160,423.698 159.371,427.251 158.215,430.555C161.655,431.206 165.046,432.438 168.249,434.287C183.544,443.118 188.792,462.705 179.962,478C173.595,489.028 161.636,494.833 149.729,493.866"
      />
    </g>
    <g
      transform="matrix(-0.5,-0.866025,0.866025,-0.5,-48.7839,287.801)"
    >
      <path
        d="M19.832,215.582C18.634,212.278 18,208.712 18,205C18,187.339 32.339,173 50,173C67.661,173 82,187.339 82,205C82,220.632 70.767,233.661 55.936,236.427"
      />
    </g>
  </svg>
</div>
`;

exports[`Logo should have the following static style rules 1`] = `
.c0 {
  float: left;
  border-radius: 999rem;
  display: block;
  z-index: 999;
  background: transparent;
}

.c0 svg {
  float: left;
}

.c0 svg {
  width: 32rem;
  height: 32rem;
}

.c0 svg > g > path {
  stroke: white;
  stroke-width: 4;
}

<div
  className="c0"
>
  <svg
    viewBox="0 0 256 256"
  >
    <g
      transform="matrix(1,0,0,1,0,-320)"
    >
      <path
        d="M99.143,406.249C104.26,395.454 115.266,388 128,388C145.661,388 160,402.339 160,420C160,423.698 159.371,427.251 158.215,430.555C161.655,431.206 165.046,432.438 168.249,434.287C183.544,443.118 188.792,462.705 179.962,478C173.595,489.028 161.636,494.833 149.729,493.866"
      />
    </g>
    <g
      transform="matrix(-0.5,-0.866025,0.866025,-0.5,-48.7839,287.801)"
    >
      <path
        d="M19.832,215.582C18.634,212.278 18,208.712 18,205C18,187.339 32.339,173 50,173C67.661,173 82,187.339 82,205C82,220.632 70.767,233.661 55.936,236.427"
      />
    </g>
  </svg>
</div>
`;

exports[`Logo should render correctly 1`] = `
.c0 {
  float: left;
  border-radius: 999rem;
  display: block;
  z-index: 999;
  background: transparent;
}

.c0 svg {
  float: left;
}

.c0 svg {
  width: 32rem;
  height: 32rem;
}

.c0 svg > g > path {
  stroke: white;
  stroke-width: 4;
}

<div
  className="c0"
>
  <svg
    viewBox="0 0 256 256"
  >
    <g
      transform="matrix(1,0,0,1,0,-320)"
    >
      <path
        d="M99.143,406.249C104.26,395.454 115.266,388 128,388C145.661,388 160,402.339 160,420C160,423.698 159.371,427.251 158.215,430.555C161.655,431.206 165.046,432.438 168.249,434.287C183.544,443.118 188.792,462.705 179.962,478C173.595,489.028 161.636,494.833 149.729,493.866"
      />
    </g>
    <g
      transform="matrix(-0.5,-0.866025,0.866025,-0.5,-48.7839,287.801)"
    >
      <path
        d="M19.832,215.582C18.634,212.278 18,208.712 18,205C18,187.339 32.339,173 50,173C67.661,173 82,187.339 82,205C82,220.632 70.767,233.661 55.936,236.427"
      />
    </g>
  </svg>
</div>
`;
